AIDevGallery/Models/ModelCompatibility.cs

```diff
@@ -20,9 +20,24 @@ private ModelCompatibility()
     public static ModelCompatibility GetModelCompatibility(ModelDetails modelDetails)
     {
         string description = string.Empty;
-
         ModelCompatibilityState compatibility;
-        if (modelDetails.HardwareAccelerators.Contains(HardwareAccelerator.CPU) ||
+
+        // check if WCR API
+        if (modelDetails.Url.StartsWith("file://", StringComparison.InvariantCultureIgnoreCase))
+        {
+            var apiKey = modelDetails.Url.Substring(7);
```

Comment:
The if should use the underlying API information:
```cs
        if (Samples.ModelTypeHelpers.ApiDefinitionDetails.Any(md => md.Value.Id == modelDetails.Id))
```
And you can retrieve the key without using Enum.Parse/TryParse:
```cs
        var apiKey = Samples.ModelTypeHelpers.ApiDefinitionDetails.FirstOrDefault(md => md.Value.Id == modelDetails.Id).Key;
```
This should be more reliable, and work for all APIs.

Reply:
done